ckan/logic/action/create.py

```diff
@@ -425,22 +425,26 @@ def resource_view_create(context, data_dict):
         model.repo.commit()
 
     # add activity for resource view create
-    user = context['user']
-    user_id = model.User.by_name(user.decode('utf8')).id
-    activity_dict = {
-        'user_id': user_id,
-        'object_id': package_id,
-        'activity_type': 'new resource view',
-        'data': {'id': resource_view.id},
-    }
-    activity_create_context = {
-        'model': model,
-        'user': user,
-        'defer_commit': False,
-        'ignore_auth': True,
-        'session': context['session'],
-    }
-    logic.get_action('activity_create')(activity_create_context, activity_dict)
+    try:
+        user = context['user']
+        user_id = model.User.by_name(user.decode('utf8')).id
```

Comment:
except block should end here, put the rest in an `else:` to avoid catching and ignoring an unrelated AttributeError